### ISSUE TITLE
Improve accessibility of form validation

### ISSUE THREAD

**Issue by luontola** [CSCfi/rems#1435]
- [x] Add "required" attribute to required for inputs
- [ ] ~~Add aria-label "required" for the asterisk (?)~~
- [x] Use fieldsets and legends to associate groups of checkboxes and radio buttons
- [x] Add aria-invalid to fields with errors
- [x] Associate error messages with the field using aria-describedby
- [x] In the error summary on the top of the page, add links to the fields with errors
- [x] In the modal popups, already the title should say whether the operation was successful or failed. Maybe add "Success:"/"Failed:" to the front of the title.
  - [ ] ~~Need aria-live regions?~~
- [x] Validation message for required attachment field
- [x] Validation message for required checkbox group

Reference: https://webaim.org/techniques/formvalidation/
